File: src/target26_2/java/com/skysoft/mixin/StatusEffectHudMixin.java
package com.skysoft.mixin;

import com.skysoft.utils.mixin.MixinErrorBoundary;
import com.skysoft.config.SkysoftConfigGui;
import net.minecraft.client.gui.Hud;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(Hud.class)
public abstract class StatusEffectHudMixin {
    @Inject(method = "extractEffects", at = @At("HEAD"), cancellable = true)
    protected void skysoftHideVanillaStatusEffects(CallbackInfo ci) {
        boolean hide = MixinErrorBoundary.value("Vanilla status effect HUD visibility", false, () -> SkysoftConfigGui.INSTANCE.config().gui.vanillaUi.areVanillaStatusEffectsHidden);
        if (hide) ci.cancel();
    }
}

File: src/target26_2/kotlin/com/skysoft/mixin/StatusEffectHudMixin.kt
package com.skysoft.mixin

import com.skysoft.config.SkysoftConfigGui
import com.skysoft.utils.SkysoftErrorBoundary
import net.minecraft.client.gui.Hud
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

@Mixin(Hud::class)
abstract class StatusEffectHudMixin {
    @Inject(method = ["extractEffects"], at = [At("HEAD")], cancellable = true)
    protected fun skysoftHideVanillaStatusEffects(ci: CallbackInfo) {
        val shouldHide = SkysoftErrorBoundary.value("Vanilla status effect HUD visibility", false) {
            SkysoftConfigGui.config().gui.areVanillaStatusEffectsHidden
        }
        if (shouldHide) ci.cancel()
    }
}
